pkg/analyze/selector/internal/relation/relation_test.go

```diff
@@ -0,0 +1,59 @@
+package relation
+
+import (
+	"testing"
+)
+
+type RelationCheckTest struct {
+	name            string
+	firstParameter  string
+	secondParameter string
+	lambda          any
+	firstArgument   any
+	secondArgument  any
+	expectedValue   bool
+	expectedError   bool
+}
+
+func TestRelationCheck(t *testing.T) {
+	tests := []RelationCheckTest{
+		{
+			name:            "simple",
+			firstParameter:  "x",
+			secondParameter: "y",
+			lambda: func(x int, y int) (bool, error) {
+				return true, nil
+			},
+			firstArgument:  1,
+			secondArgument: 2,
+			expectedValue:  true,
+			expectedError:  false,
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
Each test case should be run separately and, if possible, in parallel. For reference look at any table driven test in the repo.